src/lib.rs

```diff
@@ -1,3 +1,117 @@
+//! important for so long, and so has historical baggage, but it is also because unifying the
+//! written methods of many different cultures requires a complicated system with many parts.
+//! Unfortunately, a consequence of this is that most of the terms used in typesetting have
+//! different meanings to different people. The unicode standard is useful in that it precisely
+//! defines the terms it uses, but these meanings aren't necessarily the same as those that are in
+//! common usage.  Below are some important terms in font rendering and the meanings some ascribe
+//! to them:
+//!
+//! This is a work in progress and will be wrong in places (for now).
+//!
+//!  - *typography* - The study of drawing text to paper, screen, or some other medium. Drawing
+//!    text is also known as *typesetting*, or *rendering*.
+//!  - *symbol* - An image drawn or printed on a canvas, sheet of paper, or screen. Physically the
+//!    same as a picture.
+//!  - *glyph* - A symbol from a collection of symbols with some semantic meaning. The collection
+//!    could be a script (like latin - `ABCDE....abcde...`, or it could be from the set of
+//!    mathemtatical symbols (`+-×÷`). In the case of languages, another word more commonly used to
+//!    mean *glyph* is *letter*, but "letter" can also mean grapheme (because in latin script all
+//!    of the meanings are equivalent, they are often used interchangeably).
+//!  - *font* - A set of glyphs that can be printed to paper or a screen. Sometimes the word
+//!    font is used to mean typeface, and this can cause some confusion. In computing, a font can
+//!    also refer to 1 or more font descriptions, which contain the information required to draw
+//!    glyphs. These descriptions could be stored in a file on disk, or in memory, or somewhere
+//!    else. Popular formats include "OpenType/TrueType", "Bitmap", "PostScript", "Web Open Font
+//!    Format (WOFF)".
+//!  - *typeface* - A collection of fonts that all share the same "look", but have different sizes,
+//!    weights, and styles.
+//!  - *size* An indication of the size of a font. A font's size is often given as the width of a
+//!    latin `m` character. Sizes are often given in pixels on a computer, but screen pixels do not
+//!    have a fixed physical width. To see this think about the fact that a 14" screen with the
+//!    same number of pixels as a 21" screen will not have pixels of the same size.
+//!  - *weight* The thickness of glyphs in the font. Thicker weighted fonts are known as *bold*
+//!    whilst thinner weighted fonts are sometimes know as *light*. Weight can also be expressed as
+//!    a number.
+//!  - *style* - Variations of a specific typeface other than weight and size. Examples include
+//!    slanting text (*italics*) or using small upper case letters instead of lower case letters
+//!    (`small-caps` in CSS).
+//!  - *character* - Characters or `char`s can mean many things, but outside and within computer
```

Comment:
This is a whole rant. Here's how I would write it.

The word "character" is often used informally, but is ambiguous and has no clear formal meaning. In Rust, the `char` type actually represents a code point. In simple contexts (such as ASCII-only text), "character" can be considered identical, but in a general context it should generally be avoided if precise meaning is desired.